californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -546,6 +546,18 @@ public void receiveResponse(final Exchange exchange, final Response response) {
 					upper().receiveResponse(exchange, resp);
 				}
 			} else if (!response.hasBlockOption()) {
+				
+				// most recent none blockwise notification should stop current blockwise transfer
+				KeyUri key = getKey(exchange, response);
+				Block2BlockwiseStatus status = getBlock2Status(key);
+				if (response.getOptions().hasObserve() && status != null && status.isNotification()
+						&& status.getObserve() < response.getOptions().getObserve()) {
+					// complete current exchange
+					exchange.setComplete();
+					exchange.setCurrentRequest(exchange.getRequest());
+					// clear blockwise status
+					clearBlock2Status(key);
```

Comment:
After two days, I'm pretty sure, that I can't foresee, if the changes are proper working :-).

Just merge and check.

LGTM